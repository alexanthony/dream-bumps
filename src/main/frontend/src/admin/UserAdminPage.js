import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {Grid, Cell} from 'material-grid/dist';

import { fetchUsers } from './actions';
/**
 * UserAdminPage
 */
class UserAdminPage extends Component {
  componentWillMount() {
    this.props.fetchUsers();
  }
  render() {
    return (
      <Grid>
        <Cell col={12}>
          {this.props.children}
        </Cell>
      </Grid>
    );
  }
}

UserAdminPage.propTypes = {
  fetchUsers: PropTypes.func,
  children: PropTypes.node
}

export {UserAdminPage};

export default connect(null, { fetchUsers })(UserAdminPage);
