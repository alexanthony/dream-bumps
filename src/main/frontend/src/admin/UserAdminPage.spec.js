import React from 'react';
import { shallow } from 'enzyme';
import { UserAdminPage } from './UserAdminPage';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const fetchUsers = jest.fn();
  const component = shallow(
    <UserAdminPage
      fetchUsers={fetchUsers}
    >
      <div className="random div" />
    </UserAdminPage>
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
