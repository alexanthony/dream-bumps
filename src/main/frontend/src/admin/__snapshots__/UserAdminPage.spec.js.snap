exports[`test Renders properly 1`] = `
<Grid>
  <Cell
    col={12}>
    <div
      className="random div" />
  </Cell>
</Grid>
`;
